postgrestest.go

```diff
@@ -43,17 +51,40 @@ var ErrDisabled = errgo.New("postgres testing is disabled")
 // If the environment variable PGTESTKEEPDB is non-empty,
 // the name of the test schema will be printed and the
 // data will not be deleted.
+//
+// For optimal test performance, we recommend setting
+// the following Postgres config values in your testing
+// or development environment (BUT NEVER IN PRODUCTION):
+//
+//   fsync = off
+//   synchronous_commit = off
+//   full_page_writes = off
+//
+// Be aware that these settings may lead to data loss
+// and corruption. However, they should not have any
+// negative impact on ephemeral tests.
 func New() (*DB, error) {
-	if os.Getenv("PGTESTDISABLE") != "" {
+	if PgTestDisable() {
 		return nil, ErrDisabled
 	}
 	name := randomSchemaName()
 	db, err := sql.Open("postgres", "search_path="+name)
 	if err != nil {
 		return nil, errgo.Notef(err, "cannot open database")
 	}
-	_, err = db.Exec(`CREATE SCHEMA ` + name)
+
+	err = runWithTimeout(func(done chan error) {
```

Comment:
Why not use ExecContext and a context with a timeout here instead?

Reply:
I thought about that, but the previous code had this goroutine-based design for `DROP SCHEMA` and a comment explicitly saying that a goroutine was intentionally being used. Since this is library is used in many places, I thought it'd be safer to just follow the practice that was already in use.